101-rds-deployment-full-ha/azuredeploy.json

```diff
@@ -0,0 +1,1180 @@
+                            },
+                            "probe": {
+                                "id": "[resourceId('Microsoft.Network/loadBalancers/probes',concat(parameters('projectName'),'publb'),'rds-webgateways-probe')]"
+                            },
+                            "protocol": "Udp",
+                            "frontendPort": 3391,
+                            "backendPort": 3391,
+                            "enableFloatingIP": false,
+                            "idleTimeoutInMinutes": 5                            
+                        }
+                    }                                        
+                ]
+            }
+        },
+        {
+            "name": "[concat(parameters('projectName'),variables('sqlServerName'))]",
+            "type": "Microsoft.Sql/servers",
+            "apiVersion": "2019-06-01-preview",
+            "location": "[parameters('location')]",
+            "condition": "[parameters('createSqlServerAndDB')]",
+            "properties": {
+                "administratorLogin": "[parameters('adminUser')]",
+                "administratorLoginPassword": "[parameters('adminPasswd')]"
+            },
+            "resources": [
+                {
+                    "type": "firewallRules",
+                    "apiVersion": "2019-06-01-preview",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.Sql/servers', concat(parameters('projectName'),variables('sqlServerName')))]"
+                    ],
+                    "condition": "[parameters('createSqlServerAndDB')]",
+                    "location": "[parameters('location')]",
+                    "name": "AllowAllWindowsAzureIps",
+                    "properties": {
+                        "startIpAddress": "0.0.0.0",
+                        "endIpAddress": "0.0.0.0"
+                    }
+                }
+            ]
+        },
+        {
+            "name": "[concat(parameters('projectName'),'sqlsrv/',variables('rdsDBName'))]",
+            "type": "Microsoft.Sql/servers/databases",
+            "apiVersion": "2019-06-01-preview",
+            "location": "[parameters('location')]",
+            "condition": "[parameters('createSqlServerAndDB')]",
+            "tags": {
+                "displayName": "[concat(parameters('projectName'),'sqlsrv/',variables('rdsDBName'))]"
+            },
+            "dependsOn": [
+                "[resourceId('Microsoft.Sql/servers', concat(parameters('projectName'),'sqlsrv'))]"
+            ],
+            "properties": {
+                "collation": "SQL_Latin1_General_CP1_CI_AS",
+                "edition": "Basic",
+                "maxSizeBytes": "1073741824",
+                "requestedServiceObjectiveName": "Basic"
+            }
+        },
+        {
+            "type": "Microsoft.Resources/deployments",

```

Comment:
why is this nested?  it adds about 200 lines to the template and doesn't appear to be necessary...

Reply:
This is a nested deployment, because it has variable VM count for each type, with different parameters each. You could deploy 3 DCs, 3 Connection Brokers, 1 License server, 2 Web Access/Gateways. Each machine type will have a different Powershell DSC configuration and each machine will have its own resources.

Reply:
You can do that without nesting by applying the same conditions you have in place...  the only reason you would need this is if you were deploying to a different scope or deploying the same resourceId more than once (which doesn't appear to be the case).  But I think this is inline with the other comment - the pattern you're using for vm settings is fragile so we need to rethink that first, then can see what nesting is needed. after that...